js/cryptocom.js

```diff
@@ -0,0 +1,1207 @@
+                            [ this.parseNumber ('50000'), this.parseNumber ('0.0025') ],
+                            [ this.parseNumber ('100000'), this.parseNumber ('0.0016') ],
+                            [ this.parseNumber ('250000'), this.parseNumber ('0.00015') ],
+                            [ this.parseNumber ('1000000'), this.parseNumber ('0.00014') ],
+                            [ this.parseNumber ('20000000'), this.parseNumber ('0.00013') ],
+                            [ this.parseNumber ('100000000'), this.parseNumber ('0.00012') ],
+                            [ this.parseNumber ('200000000'), this.parseNumber ('0.0001') ],
+                        ],
+                    },
+                },
+            },
+            'options': {},
+            // https://exchange-docs.crypto.com/spot/index.html#response-and-reason-codes
+            'exceptions': {
+                'exact': {
+                    '10001': ExchangeError,
+                    '10002': PermissionDenied,
+                    '10003': PermissionDenied,
+                    '10004': BadRequest,
+                    '10005': PermissionDenied,
+                    '10006': DDoSProtection,
+                    '10007': InvalidNonce,
+                    '10008': BadRequest,
+                    '10009': BadRequest,
+                    '20001': BadRequest,
+                    '20002': InsufficientFunds,
+                    '30003': BadSymbol,
+                    '30004': BadRequest,
+                    '30005': BadRequest,
+                    '30006': BadRequest,
+                    '30007': BadRequest,
+                    '30008': BadRequest,
+                    '30009': BadRequest,
+                    '30010': BadRequest,
+                    '30013': BadRequest,
+                    '30014': BadRequest,
+                    '30016': BadRequest,
+                    '30017': BadRequest,
+                    '30023': BadRequest,
+                    '30024': BadRequest,
+                    '30025': BadRequest,
+                    '40001': BadRequest,
+                    '40002': BadRequest,
+                    '40003': BadRequest,
+                    '40004': BadRequest,
+                    '40005': BadRequest,
+                    '40006': BadRequest,
+                    '40007': BadRequest,
+                    '50001': BadRequest,
+                },
+            },
+        });
+    }
+
+    nonce () {
+        return this.milliseconds ();
+    }
+
+    sign (path, api = 'public', method = 'GET', params = {}, headers = undefined, body = undefined) {
+        let url = this.urls['api'] + '/' + path;
+        const query = this.omit (params, this.extractParams (path));
+        if (api === 'public') {
+            if (Object.keys (query).length) {
+                url += '?' + this.urlencode (query);
+            }
+        } else {
+            this.checkRequiredCredentials ();
+            const id = '1';
```

Comment:
Setting this id to a constant value isn't really correct.
We have to use `this.milliseconds()` or something similar here.
Their docs say it must be a long value, not a string.